### PR TITLE
QKSMS updates

### PR DESCRIPTION
**Pull request by jeb218**
1. Fixed forwarding messages with attachments (Issue #62)
2. Fixed texting to email addresses  (Issue #653)
3. Fixed incoming slideshows not showing all images.  (Issue #498 )
4. Added support for multiple images being sent, (RecyclerView horizontal slider), and fixed slideshow code for outgoing messages
5. Changed avatars for multiple recipients on a conversation. It now shows all the avatars for people in the group (up to 4)

### CODE REVIEW

**Comment by moezbhatti** [comment]
Just wanted to say, before I've even looked at the code, this is incredible. These are some really big contributions!

**Comment by jeb218** [comment]
I hope that it can help out, and that I can contribute more in the future. There are a couple things in what I wrote that isn't perfect. There's some redundancy in what I did to fix the slideshow section, I meant to go back and clean it up a bit but haven't yet.

File: QKSMS/src/main/java/com/moez/QKSMS/model/SlideshowModel.java [hunk 1]
```diff
@@ -223,20 +246,32 @@ public static SlideshowModel createFromPduBody(Context context, PduBody pb) thro
...
-            slides.add(slide);
+                SlideModel slide = new SlideModel((int) (par.getDur() * 1000), mediaSet);
+                slide.setFill(par.getFill());
+                SmilHelper.addParElementEventListeners((EventTarget) par, slide);
+
+                if(sme.getTagName().equals(SmilHelper.ELEMENT_TAG_TEXT)){
+                    body = slide.getText();
+                    continue;
+                }
+                else{
```
Comment (partounian) [inline QKSMS/src/main/java/com/moez/QKSMS/model/SlideshowModel.java]: Else is incorrectly formatted here, should be "} else {" like the rest. 👍

Comment (partounian) [inline QKSMS/src/main/java/com/moez/QKSMS/model/SlideshowModel.java]: Also thank you very much for the changes, I hope we can get it to pass the tests soon.

**Comment by moezbhatti** [comment]
Going to close this since the new version of the app is using a completely re-written codebase. Sorry I wasn't very proactive about getting this merged in, I'll try to be a lot better at that moving forward

**Pull request closed without merging**